sources/packages/api/src/services/disbursement-schedule-service/disbursement-schedule-service.ts

```diff
@@ -305,4 +329,127 @@ export class DisbursementScheduleService extends RecordDataModelService<Disburse
     coeQuery.orderBy("disbursementSchedule.coeStatus");
     return coeQuery.getMany();
   }
+
+  /**
+   * Returns Disbursement and application details for COE Application details.
+   * @param disbursementScheduleId
+   * @returns Disbursement and Application details.
+   */
+  async getDisbursementAndApplicationDetails(
+    locationId: number,
+    disbursementScheduleId: number,
+  ): Promise<DisbursementSchedule> {
+    return this.repo
+      .createQueryBuilder("disbursementSchedule")
+      .select([
+        "disbursementSchedule.id",
+        "disbursementSchedule.disbursementDate",
+        "disbursementSchedule.coeStatus",
+        "disbursementSchedule.application",
+        "disbursementSchedule.coeDeniedOtherDesc",
+      ])
+      .innerJoinAndSelect("disbursementSchedule.application", "application")
```

Comment:
i think we decided not to use `innerJoinAndSelect`.

Reply:
It would be better to select only what is needed from application, as @ann-aot mentioned.

Reply:
Yes. Actually I was moving this code from application service to here. So just moved most of the logic without changing it.

Reply:
Modified the query without using joinAndSelect.